pkg/cfn/builder/nodegroup.go

```diff
@@ -110,7 +110,7 @@ func (n *NodeGroupResourceSet) addResourcesForNodeGroup() error {
 	launchTemplateName := gfn.MakeFnSubString(fmt.Sprintf("${%s}", gfn.StackName))
 	launchTemplateData := newLaunchTemplateData(n)
 
-	if api.IsEnabled(n.spec.SSH.Allow) && api.IsSetAndNonEmptyString(n.spec.SSH.PublicKeyName) {
```

Comment:
Even though we set proper values to this field it might be good to be cautious and check for `n.spec.SSH` being `nil` before checking the publicKeyName.

Reply:
Fair call, I've added that change now :)